src/meta-srv/src/procedure/region_migration/update_metadata.rs

```diff
@@ -39,30 +41,51 @@ pub enum UpdateMetadata {
     Rollback,
 }
 
+fn clean_lock_keys(procedure_ctx: &ProcedureContext, key: &StringKey, guard: OwnedKeyRwLockGuard) {
+    drop(guard);
+    procedure_ctx.provider.clean_lock_keys(key);
+}
```

Comment:
Is it possible to use RAII style to release the lock? Can we wrap a struct to hold the guard?

Reply:
resolved in [80139e5](https://github.com/GreptimeTeam/greptimedb/pull/6023/commits/80139e5c21a8033e864954549847cd756ebb7332)